### PR TITLE
feat: Load site functions via RemoteConfig

### PR DESCRIPTION
**Pull request by benjackwhite**
## Changes

Needs https://github.com/PostHog/posthog-js/pull/1581
Needs https://github.com/PostHog/posthog/pull/26671

- [x] Moves the site functions to _only_ be loaded the new way
- [x] Changes the loading mentality so that the caller is responsible for initializing and passing things in (this is pretty important as we don't want the server side code to be calling non-top-level methods as they might change over time).

## Checklist
- [ ] Tests for new code (see [advice on the tests we use](https://github.com/PostHog/posthog-js#tiers-of-testing))
- [ ] Accounted for the impact of any changes across different browsers
- [ ] Accounted for backwards compatibility of any changes (no breaking changes in posthog-js!)

### CODE REVIEW

File: playground/nextjs/src/posthog.ts [hunk 1]
```diff
@@ -62,10 +62,12 @@ if (typeof window !== 'undefined') {
         persistence: cookieConsentGiven() ? 'localStorage+cookie' : 'memory',
         person_profiles: PERSON_PROCESSING_MODE === 'never' ? 'identified_only' : PERSON_PROCESSING_MODE,
         persistence_name: `${process.env.NEXT_PUBLIC_POSTHOG_KEY}_nextjs`,
+        opt_in_site_apps: true,
         __preview_remote_config: true,
         ...configForConsent(),
     })
-    // Help with debugging(window as any).posthog = posthog
+    // Help with debugging
+    ;(window as any).posthog = posthog
```
Comment (benjackwhite) [inline playground/nextjs/src/posthog.ts:70]: Decide this should always be on

File: src/types.ts [hunk 2]
```diff
@@ -341,6 +341,7 @@ export interface PostHogConfig {
 
     /**
      * PREVIEW - MAY CHANGE WITHOUT WARNING - DO NOT USE IN PRODUCTION
+     * whether to wrap fetch and add tracing headers to the request
```
Comment (benjackwhite) [inline src/types.ts:344]:
```suggestion
     * enables the new RemoteConfig approach to loading config instead of decide
```

**Review by mariusandra — APPROVED** [review]
Looks good. The only question I had was if legacy site apps still work, but reading the code more it seems like that's handled. So ✅

**Comment by benjackwhite** [comment]
> Looks good. The only question I had was if legacy site apps still work, but reading the code more it seems like that's handled. So ✅

They _should_ do. The actual path for loading them is basically identical to before. I'll test again for sure before merging thouh.

**Pull request merged**